bears/general/InvalidLinkBear.py

```diff
@@ -40,18 +40,32 @@ def get_status_code(url, timeout):
             pass
 
     @staticmethod
+    def parse_pip_vcs_url(link):
+        splitted_at = link.split('@')[0]
+        splitted_tag = splitted_at.split('#')[0]
+        splitted_schema = splitted_tag[splitted_tag.index('+') + 1:]
+        return splitted_schema
+
+    @staticmethod
     def find_links_in_file(file, timeout, link_ignore_regex):
         link_ignore_regex = re.compile(link_ignore_regex)
         regex = re.compile(
-            r'(https?://[^.:%\s_/?#[\]@\\]+\.(?:[^\s()%\'"`<>|\\]+|'
-            r'\([^\s()%\'"`<>|\\]*\))*)(?<!\.)(?<!,)')
+            r'((git\+|bzr\+|svn\+|hg\+)?https?://[^.:%\s_/?#[\]@\\]+\.(?:[^\s('
+            r')%\'"`<>|\\]+|\([^\s()%\'"`<>|\\]*\))*)(?<!\.)(?<!,)')
         for line_number, line in enumerate(file):
             match = regex.search(line)
             if match:
                 link = match.group()
                 if not link_ignore_regex.search(link):
-                    code = InvalidLinkBear.get_status_code(link, timeout)
-                    yield line_number + 1, link, code
+                    if ('hg+' in link) or ('bzr+' in link) or \
+                       ('git+' in link) or ('svn+' in link):
+                        parsed_link = InvalidLinkBear.parse_pip_vcs_url(link)
```

Comment:
I'd do `link = Inva...` then you can unindent line 63ff again and remove the else clause, less code duplication